tss/node/signer/verify.go

```diff
@@ -24,24 +26,34 @@ func (p *Processor) Verify() {
 			case <-p.stopChan:
 				return
 			case req := <-p.askRequestChan:
-				var askRequest tsstypes.SignStateRequest
+				var askRequest common.SignStateRequest
+				var RpcResponse tdtypes.RPCResponse
 				if err := json.Unmarshal(req.Params, &askRequest); err != nil {
 					logger.Error().Msg("failed to unmarshal ask request")
+					RpcResponse = tdtypes.NewRPCErrorResponse(req.ID, 201, "failed to unmarshal ", err.Error())
+					p.wsClient.SendMsg(RpcResponse)
 					continue
 				}
 				var resId = req.ID
+				offset, err := strconv.Atoi(askRequest.OffsetStartsAtIndex)
+				if err != nil {
+					logger.Error().Msg("failed to conv OffsetStartsAtIndex to int")
+					RpcResponse = tdtypes.NewRPCErrorResponse(req.ID, 201, "failed to conv OffsetStartsAtIndex to int", err.Error())
+					p.wsClient.SendMsg(RpcResponse)
+					continue
+				}
 
 				// if offset value does not equal  length of state roots array, return false
-				if len(askRequest.StateRoots) != int(askRequest.OffsetStartsAtIndex.Int64()) {
-					askResponse := tsstypes.AskResponse{
+				if len(askRequest.StateRoots) != offset {
```

Comment:
plz remove this check. offset has no concern with the number of askRequest.StateRoots